src/memory-analyzer/gdb_api.cpp

```diff
@@ -0,0 +1,474 @@
+/*******************************************************************\
+
+Module: GDB Machine Interface API
+
+Author: Malte Mues <mail.mues@gmail.com>
+        Daniel Poetzl
+
+\*******************************************************************/
+
+/// \file
+/// Low-level interface to gdb
+
+#ifdef __linux__
+#include <cctype>
+#include <cerrno>
+#include <cstdio>
+#include <cstring>
+#include <regex>
+
+#include <iostream>
+
+#include "gdb_api.h"
+
+#include <goto-programs/goto_model.h>
+
+#include <util/string_utils.h>
+
+/// Create a gdb_apit object
+///
+/// \param binary the binary to run with gdb
+/// \param log boolean indicating whether gdb input and output should be logged
+gdb_apit::gdb_apit(const char *binary, const bool log)
+  : binary(binary), log(log), gdb_state(gdb_statet::NOT_CREATED)
+{
+}
+
+/// Create a new gdb process for analysing the binary indicated by the member
+/// variable `binary`
+void gdb_apit::create_gdb_process()
+{
+  PRECONDITION(gdb_state == gdb_statet::NOT_CREATED); std::cout << "GDB Here 0" << std::endl;
```

Comment:
I guess this needs some cleanup :-) You might as well use a `debug()` `mstreamt`, possibly with a separate debug logger so that you can use, e.g., a stream-based message handler that puts all the debug output on `stderr`?

Reply:
Ah yes I just added this as I'm debugging some CI issues. This will be removed later.